curriculum/challenges/english/08-coding-interview-prep/rosetta-code/self-referential-sequence.english.md

```diff
@@ -0,0 +1,151 @@
+---
+id: 5eb3e4a21f462f409d656c73
+title: Self-referential sequence
+challengeType: 5
+forumTopicId: 385317
+---
+
+## Description
+
+<section id='description'>
+There are several ways to generate a self-referential sequence. One very common one (the <a href="https://rosettacode.org/wiki/Look-and-say sequence" target="_blank">Look-and-say sequence</a>) is to start with a positive integer, then generate the next term by concatenating enumerated groups of adjacent alike digits:
```

Comment:
We might want to add what a self-referential sequence is

Reply:
I did not find a formal definition but mostly its just the essence of the words "self-referential sequence" which we can describe. We can add this line: A self-referential sequence is a sequence whose elements are generated by using its current elements.
What do you think?

Reply:
I guess rather than coming up with one of our own, we should leave it to the user to understand in that case?

Reply:
@thecodingaviator, I think it's okay to let learners look up the definition or go to the Look-and-say sequence to get a better understanding of what a self-referential sequence is. Also if they read on they get some concrete examples.